src/EditorFeatures/Core/InlineParameterNameHints/InlineParamNameHintsDataTaggerProvider.cs

```diff
@@ -0,0 +1,61 @@
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.ComponentModel.Composition;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Editor.Shared.Extensions;
+using Microsoft.CodeAnalysis.Editor.Shared.Tagging;
+using Microsoft.CodeAnalysis.Editor.Shared.Utilities;
+using Microsoft.CodeAnalysis.Editor.Tagging;
+using Microsoft.CodeAnalysis.InlineParameterNameHints;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Shared.TestHooks;
+using Microsoft.CodeAnalysis.Text.Shared.Extensions;
+using Microsoft.VisualStudio.Text;
+using Microsoft.VisualStudio.Text.Editor;
+using Microsoft.VisualStudio.Text.Tagging;
+using Microsoft.VisualStudio.Utilities;
+
+namespace Microsoft.CodeAnalysis.Editor.InlineParameterNameHints
+{
+    /// <summary>
+    /// The TaggerProvider that calls upon the service in order to locate the spans and names
+    /// </summary>
+    [Export(typeof(ITaggerProvider))]
+    [ContentType("csharp")]
+    [TagType(typeof(InlineParamNameHintDataTag))]
+    [Name(nameof(InlineParamNameHintsDataTaggerProvider))]
+    internal class InlineParamNameHintsDataTaggerProvider : AsynchronousTaggerProvider<InlineParamNameHintDataTag>
+    {
+        [ImportingConstructor]
+        public InlineParamNameHintsDataTaggerProvider(
+            IThreadingContext threadingContext,
+            IAsynchronousOperationListenerProvider listenerProvider,
+            IForegroundNotificationService notificationService)
+            : base(threadingContext, listenerProvider.GetListener(FeatureAttribute.InlineParameterNameHints), notificationService)
+        {
+        }
+
+        protected override ITaggerEventSource CreateEventSource(ITextView textViewOpt, ITextBuffer subjectBuffer)
+        {
+            return TaggerEventSources.OnTextChanged(subjectBuffer, TaggerDelay.NearImmediate);
+        }
+
+        protected override async Task ProduceTagsAsync(TaggerContext<InlineParamNameHintDataTag> context, DocumentSnapshotSpan documentSnapshotSpan, int? caretPosition)
+        {
+            var cancellationToken = context.CancellationToken;
+            var document = documentSnapshotSpan.Document;
+
+            var snapshotSpan = documentSnapshotSpan.SnapshotSpan;
+            var paramNameHintsService = document.GetLanguageService<IInlineParamNameHintsService>();
+            var paramNameHintSpans = await paramNameHintsService.GetInlineParameterNameHintsAsync(document, snapshotSpan.Span.ToTextSpan(), cancellationToken).ConfigureAwait(false);
+            cancellationToken.ThrowIfCancellationRequested();
+
+            foreach (var span in paramNameHintSpans)
+            {
+                context.AddTag(new TagSpan<InlineParamNameHintDataTag>(span.Span.ToSnapshotSpan(snapshotSpan.Snapshot), new InlineParamNameHintDataTag(span.Name))); //span.Item1.Length + 1, _format); //));
```

Comment:
Probably worth putting a cancellationToken check here in the foreach since this could potentially get long.